src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -10,6 +10,8 @@ namespace AppInstaller::Fonts
 {
     namespace
     {
+        std::vector<std::wstring> preferredLocales = AppInstaller::Locale::GetUserPreferredLanguagesUTF16();
```

Comment:
A global static is not how we should be reducing the number of calls to `GetUserPreferredLanguagesUTF16`.  Global statics bring along several issues, made even worse when they are not statically initialized.

A better approach would be to have the font enumeration functions actually be methods of a type.  The caller then controls the lifetime of anything that the font enumeration caches, like a member field of locales.

Reply:
Created a struct called `FontCatalog` that hold all of these font enumeration methods as well as a private member field for the preferred locales.